jina/peapods/peas/__init__.py

```diff
@@ -179,6 +180,10 @@ def close(self) -> None:
             # if it is not daemon, block until the process/thread finish work
             if not self.args.daemon:
                 self.join()
+        else:
+            # if it fails to start, the process will hang at `join`
+            if getattr(self, 'terminate', None):
```

Comment:
I can't find where terminate is defined

Reply:
in the Process parent class if backend is Process.